lib/lhm/sql_retry.rb

```diff
@@ -57,8 +165,29 @@ def default_retry_config
         rand_factor: 0, # percentage to randomize the next retry interval time
         max_elapsed_time: Float::INFINITY, # max total time in seconds that code is allowed to keep being retried
         on_retry: Proc.new do |exception, try_number, total_elapsed_time, next_interval|
-          log = "#{exception.class}: '#{exception.message}' - #{try_number} tries in #{total_elapsed_time} seconds and #{next_interval} seconds until the next try."
-          log_with_prefix(log, :info)
+          if reconnect_successful?(exception)
+            log_with_prefix("#{exception.message} -- triggering retry", :info)
+          else
+            log_with_prefix("#{exception.class}: '#{exception.message}' - #{try_number} tries in #{total_elapsed_time} seconds and #{next_interval} seconds until the next try.", :error)
+          end
+        end
+      }.freeze
+    end
+
+    def host_retry_config
+      {
+        on: {
+          StandardError => [
+            /Lost connection to MySQL server at 'reading initial communication packet'/
+          ]
+        },
+        multiplier: 1, # each successive interval grows by this factor
+        base_interval: 0.25, # the initial interval in seconds between tries.
+        tries: 20, # Number of attempts to make at running your code block (includes initial attempt).
+        rand_factor: 0, # percentage to randomize the next retry interval time
```

Comment:
Coarsly tuned. Taken from the "default_retry_config", but the base_interval was changed to 0.25s, which means it will try to reconnect for 5 seconds. What do you think these numbers should be?

Reply:
But what do you think? Is 5s of retries good enough to overcome the most common network blips? Especially since this will go on top of ActiveRecord's own retry logic